d2bs/kolbot/libs/bots/BoBot.js

```diff
@@ -0,0 +1,230 @@
+/**
+*	@filename	BoBot.js
+*	@author		casualBotter
+*	@desc		Meant to run alongside Enchant to Bo players
+*/
+
+function BoBot(){
+	var command, hostile, nick, spot, tick, s, m,
+		startTime = getTickCount(),
+		shitList = [],
+		greet = [];
+
+		this.Bo = function (nick) {
+		if (!Misc.inMyParty(nick)) {
+			say("Accept party invite, noob.");
+
+			return false;
+		}
+
+		var partyUnit,
+			unit = getUnit(0, nick);
+
+		if (getDistance(me, unit) > 10) {
+			say("Get closer.");
+
+			return false;
+		}
+
+		if (!unit) {
+			partyUnit = getParty(nick);
+
+			// wait until party area is readable?
+
+			if ([1,40, 75, 103, 109].indexOf(partyUnit.area) > -1) {
+				say("Cannot Bo in town.");
+				print("cannot bo in town");
+				return false
+			} else{
+				print(partyUnit.area); //for debug
+				Pather.useWaypoint(partyUnit.area);

```

Comment:
Should check if `partyUnit.area` has actually a waypoint. If not you could handle the case by telling the player he should get close to a waypoint.

Reply:
Yeah you're right I added a check for that and will update when I figure out distance issue